### PR TITLE
Rewrites anti-magic slightly to stop spamming anti-magic effects on the same mob to silly levels

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Moves anti-magic effect off of the component and onto the mob. Fixes some bugs while there. 

Anti-magic effects will now only trigger on calls with a charge cost associated. 

If the user is under the effects of.. an antimagic effect, a new one will not play. This cuts down on overlay and sound effect spam from walking through forcewall or spacetime distortion, for example. 

## Why It's Good For The Game

I just found it a little annoying to see a chaplain walk through a magic effect and it just spams the fart sound over and over on repeat and makes them completely impossible to see. 

## Changelog

:cl: Melbert
fix: Anti-magic visual effects will no longer trigger overtop one another
/:cl:

### CODE REVIEW

File: code/datums/components/anti_magic.dm [hunk 1]
```diff
@@ -36,17 +38,14 @@
...
 	)
 
 	if(isitem(parent))
 		RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, PROC_REF(on_equip))
 		RegisterSignal(parent, COMSIG_ITEM_DROPPED, PROC_REF(on_drop))
 	else if(ismob(parent))
-		RegisterSignal(parent, COMSIG_MOB_RECEIVE_MAGIC, PROC_REF(block_receiving_magic), override = TRUE)
-		RegisterSignal(parent, COMSIG_MOB_RESTRICT_MAGIC, PROC_REF(restrict_casting_magic), override = TRUE)
-		if(!HAS_TRAIT(parent, TRAIT_ANTIMAGIC_NO_SELFBLOCK))
-			to_chat(parent, span_warning("Magic seems to flee from you. You are immune to spells but are unable to cast magic."))
```
Comment (timothymtorres) [inline code/datums/components/anti_magic.dm:49]: You removed the antimagic warning.  Was this spamming the chat or something?

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -979,16 +979,65 @@
...
+
 	return is_magic_blocked
 
+/// Called whenever a magic effect with a charge cost is blocked and we haven't recently blocked magic.
+/mob/proc/on_block_magic_effects(magic_flags, list/antimagic_sources)
+	return
+
+/mob/living/on_block_magic_effects(magic_flags, list/antimagic_sources)
+	ADD_TRAIT(src, TRAIT_RECENTLY_BLOCKED_MAGIC, MAGIC_TRAIT)
+	addtimer(TRAIT_CALLBACK_REMOVE(src, TRAIT_RECENTLY_BLOCKED_MAGIC, MAGIC_TRAIT), 6 SECONDS)
```
Comment (timothymtorres) [inline code/modules/mob/mob.dm:1004]:
```suggestion
	addtimer(TRAIT_CALLBACK_REMOVE(src, TRAIT_RECENTLY_BLOCKED_MAGIC, MAGIC_TRAIT), 6 SECONDS, TIMER_UNIQUE|TIMER_OVERRIDE)
```

Comment (timothymtorres) [reply code/modules/mob/mob.dm:1004]: Just reuse the same timer and reset the cooldown.

Comment (MrMelbert) [reply code/modules/mob/mob.dm:1004]: The cooldown would never actually reset since this proc is not run if they have the trait

Comment (timothymtorres) [reply code/modules/mob/mob.dm:1004]: Missed that.  Then it doesn't matter if we reset the timer or not since it only gets used once.

**Comment by timothymtorres** [comment]
All the changes look good!  After testing I came across an already existing bug that doesn't update the spell UI buttons:

https://github.com/tgstation/tgstation/blob/a6b2c054b935bc99beefef4396d5fc85f9948787/code/modules/spells/spell.dm#L105-L106

Just add `RegisterSignal(owner, COMSIG_MOB_UNEQUIPPED_ITEM, PROC_REF(update_status_on_signal))` and then remove the signal in the `UnregisterSignal` list.

**Pull request merged**